### PR TITLE
Dan/plat 681 modify previewjson format

### PR DESCRIPTION
**Pull request by dramich**
Refactor the PreviewConfig format and maintain existing functionality

### CODE REVIEW

File: internal/api/preview_config.go [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
-	PackageParams map[string]interface{} `json:"packageParams"`
+	ProjectSlug string                    `json:"projectSlug"`
+	Credentials []Credential              `json:"credentials"`
+	Packages    map[string]PreviewPackage `json:"packages"`
+}
+
+type PreviewPackage struct {
+	Params           map[string]interface{} `json:"params"`
+	Secrets          []Secret               `json:"secrets,omitempty"`
+	RemoteReferences []RemoteRef            `json:"remoteReferences,omitempty"`
```
Comment (akasprzok) [inline internal/api/preview_config.go:12]: Note these are all maps now, not lists.

See https://github.com/massdriver-cloud/massdriver/pull/2047.

File: internal/api/preview_config.go [hunk 2]
```diff
@@ -1,19 +1,26 @@
 package api
 
 type PreviewConfig struct {
-	ProjectSlug   string                 `json:"projectSlug"`
-	Credentials   map[string]string      `json:"credentials"`
-	PackageParams map[string]interface{} `json:"packageParams"`
+	ProjectSlug string                    `json:"projectSlug"`
```
Comment (akasprzok) [inline internal/api/preview_config.go:4]: We don't need ProjectSlug, the project ID is in the query already.

Comment (dramich) [reply internal/api/preview_config.go:4]: It is needed, this is the definition of the JSON file and the source of the slug/id when sending the request over.

Comment (akasprzok) [reply internal/api/preview_config.go:4]: this is what the mutation currently looks like:

```
mutation deployPreviewEnvironment($organizationId: ID!, $projectId: ID!, $packageConfigurations: JSON!, $ciContext: JSON!, $credentials: JSON!) {
        deployPreviewEnvironment(
          organizationId: $organizationId,
          projectId: $projectId,
          input: {
            credentials: $credentials
            packageConfigurations: $packageConfigurations,
            ciContext: $ciContext,
          }
          ) {
          result {
            name
            slug
            mode
          }
        }
      }
```
      
 We currently have the field in there as `projectId`.

**Review by akasprzok — APPROVED** [review]

**Pull request merged**